addons/account/tests/test_account_fiscal_year.py

```diff
@@ -0,0 +1,151 @@
+        :param expected_date_from: The expected date_from after computation.
+        :param expected_date_to: The expected date_to after computation.
+        '''
+        current_date = datetime.strptime(date, DEFAULT_SERVER_DATE_FORMAT)
+        res = company.compute_fiscalyear_dates(current_date)
+        self.assertEqual(res['date_from'].strftime(DEFAULT_SERVER_DATE_FORMAT), expected_date_from)
+        self.assertEqual(res['date_to'].strftime(DEFAULT_SERVER_DATE_FORMAT), expected_date_to)
+
+    def test_default_fiscal_year(self):
+        '''Basic case with a fiscal year xxxx-01-01 - xxxx-12-31.'''
+        company = self.env.ref('base.main_company')
+        company.fiscalyear_last_day = 31
+        company.fiscalyear_last_month = 12
+
+        self.check_compute_fiscal_year(
+            company,
+            '2017-12-31',
+            '2017-01-01',
+            '2017-12-31',
+        )
+
+        self.check_compute_fiscal_year(
+            company,
+            '2017-01-01',
+            '2017-01-01',
+            '2017-12-31',
+        )
+
+    def test_leap_fiscal_year_1(self):
+        '''Case with a leap year ending the 29 February.'''
+        company = self.env.ref('base.main_company')
+        company.fiscalyear_last_day = 29
+        company.fiscalyear_last_month = 2
+
+        self.check_compute_fiscal_year(
+            company,
+            '2016-02-29',
+            '2015-03-01',
+            '2016-02-29',
+        )
+
+    def test_leap_fiscal_year_2(self):
+        '''Case with a next leap year ending the 29 February.'''
+        company = self.env.ref('base.main_company')
+        company.fiscalyear_last_day = 29
+        company.fiscalyear_last_month = 2
+
+        self.check_compute_fiscal_year(
+            company,
+            '2015-03-01',
+            '2015-03-01',
+            '2016-02-29',
+        )
+
+    def test_leap_fiscal_year_3(self):
+        '''Case with a leap year ending the 28 February.'''
+        company = self.env.ref('base.main_company')
+        company.fiscalyear_last_day = 28
+        company.fiscalyear_last_month = 2
+
+        self.check_compute_fiscal_year(
+            company,
+            '2016-03-01',
+            '2016-02-29',
+            '2017-02-28',
+        )
+
+    def test_custom_fiscal_year_1(self):
+        '''Case with a custom fiscal year covering the six first months of the year.'''
+        company = self.env.ref('base.main_company')
+        company.fiscalyear_last_day = 31
+        company.fiscalyear_last_month = 12
+
+        self.env['account.fiscal.year'].create({
+            'name': '6 month 2017',
+            'date_from': '2017-01-01',
+            'date_to': '2017-05-31',
+            'company_id': company.id,
+        })
+
+        self.check_compute_fiscal_year(
+            company,
+            '2017-11-01',
+            '2017-06-01',
+            '2017-12-31',
+        )
+
+    def test_custom_fiscal_year_2(self):
```

Comment:
merge test_custom_fiscal_year_1 and test_custom_fiscal_year_2

and test_custom_fiscal_year_3 (create the first 'account.fiscal.year' record, then check dates '2017-11-01' and '2017-02-01' then create second 'account.fiscal.year' record, and finally check '2017-07-01')